### PR TITLE
User setup

### LINKED ISSUE #23: User backend setup
**Describe the task that needs to be done.**
Setup User entity, DTO, repository, service and controller. 

**Describe how a solution to your proposed task might look like (and any alternatives considered).**
User entity fields:
- id: UUID 
- name: string 
- email: string 
- pair: Pair
- authCookie: String/Cookie

Endpoints: 

Retrieving users
GET /users/{id}
Response:
200 OK w/ User Entity

**Notes**

### PR DESCRIPTION
**Pull request by uoa-ypec413**
## Description
Sets up entities, DTOs, repository, service, and controller for the user endpoint(s).
Fixes #23 

## How has this been tested?

- Created and ran unit tests for the service and controller. 
- Tested on local Swagger UI, endpoint shows up.
- Tested local H2 console and viewed created tables.

## Checklist
*(Leave blank if not applicable)*

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -1,3 +1,6 @@
 spring.datasource.driverClassName=org.h2.Driver
 spring.h2.console.enabled=true
 spring.h2.console.path=/h2-console
+spring.datasource.url=jdbc:h2:mem:testdb
```
Comment (uoa-ypec413) [inline src/main/resources/application.properties:4]: These lines were added due to an issue with accessing the local H2 console, where username and password were incorrect.
unknown

File: src/main/java/com/team701/buddymatcher/domain/users/User.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    public User setEmail(String email) {
+        this.email = email;
+        return this;
+    }
+
+    public Pair getPair() {
+        return pair;
+    }
+
+    public User setPair(Pair pair) {
```
Comment (JafarMaash) [inline src/main/java/com/team701/buddymatcher/domain/users/User.java:55]: Is it worth maybe checking that the pair being passed in contains this user? Not sure if that type of error could ever come up though so idk

Comment (uoa-ypec413) [reply src/main/java/com/team701/buddymatcher/domain/users/User.java:55]: I think this check would be useful for any future endpoints or services we create that set or update the user's pair, but I personally prefer to keep the logic for Entity setters/getters as simple as possible.

**Review by JafarMaash — APPROVED** [review]
LGTM :)

File: src/main/java/com/team701/buddymatcher/domain/users/User.java (hunk 2 above)
Comment (JafarMaash) [reply src/main/java/com/team701/buddymatcher/domain/users/User.java:55]: That makes sense, sounds good

**Pull request merged**